### PR TITLE
[Merged by Bors] - chore: Remove order dependencies to `Data.Fin.Basic`

### PR DESCRIPTION
**Pull request by YaelDillies**
Move all the declarations using the mathlib order hierarchy to a new file `Order.Fin`. There were a bunch of order embeddings in `Data.Fin.Basic`, so I kept an embedding version of them around and renamed the order embedding version.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Fin.lean [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+Authors: Robert Y. Lewis, Keeley Hoek
+-/
+import Mathlib.Data.Fin.Basic
+import Mathlib.Order.Hom.Set
+
+#align_import data.fin.basic from "leanprover-community/mathlib"@"3a2b5524a138b5d0b818b858b516d4ac8a484b03"
+
+/-!
+# `Fin n` forms a ounded linear order
+
```
Comment (dupuisf) [inline Mathlib/Order/Fin.lean:13]:
```suggestion
# `Fin n` forms a founded linear order

```

**Review by dupuisf — COMMENTED** [review]
Should we add an `assert_not_exists` statement in Data/Fin/Basic?

**Comment by YaelDillies** [comment]
> Should we add an `assert_not_exists` statement in Data/Fin/Basic?

There's no meaningul `assert_not_exists` statement to write before @Ruben-VandeVelde's #13092.

**Pull request closed without merging**